system-contracts/contracts/Compressor.sol

```diff
@@ -48,24 +49,27 @@ contract Compressor is ICompressor, ISystemContract {
         unchecked {
             (bytes calldata dictionary, bytes calldata encodedData) = _decodeRawBytecode(_rawCompressedData);
 
-            require(
-                encodedData.length * 4 == _bytecode.length,
-                "Encoded data length should be 4 times shorter than the original bytecode"
-            );
+            if (encodedData.length * 4 != _bytecode.length) {
+                revert MalformedBytecode(BytecodeError.Length);
+            }
 
-            require(
-                dictionary.length / 8 <= encodedData.length / 2,
-                "Dictionary should have at most the same number of entries as the encoded data"
-            );
+            if (dictionary.length / 8 > encodedData.length / 2) {
+                revert MalformedBytecode(BytecodeError.DictionaryLength);
+            }
 
+            // solhint-disable-next-line gas-length-in-loops
             for (uint256 encodedDataPointer = 0; encodedDataPointer < encodedData.length; encodedDataPointer += 2) {
                 uint256 indexOfEncodedChunk = uint256(encodedData.readUint16(encodedDataPointer)) * 8;
-                require(indexOfEncodedChunk < dictionary.length, "Encoded chunk index is out of bounds");
+                if (indexOfEncodedChunk > dictionary.length - 1) {
+                    revert IndexOutOfBounds();
+                }
 
                 uint64 encodedChunk = dictionary.readUint64(indexOfEncodedChunk);
                 uint64 realChunk = _bytecode.readUint64(encodedDataPointer * 4);
 
-                require(encodedChunk == realChunk, "Encoded chunk does not match the original bytecode");
+                if (encodedChunk != realChunk) {
+                    revert ValuesNotEqual(realChunk, encodedChunk);
```

Comment:
Which values? need to make a specific error for compressor encoded chunks rather than reuse `ValuesNotEqual` imho